### PR TITLE
Updated to Swift 3

### PR DESCRIPTION
**Pull request by ChristopherBell**
Most errors fixed but still a few remaining.

### CODE REVIEW

File: Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift [hunk 1]
```diff
@@ -53,7 +53,7 @@ class MoneyPile {
         if v == 0 {
             return true
         } else if let next = self.nextPile {
-            return next.canWithdraw(v)
+            return next.canWithdraw(v: v)
```
Comment (ochococo) [inline Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift]: We should avoid parameters like `v`. May I suggest `amount:` ?

File: Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift [hunk 2]
```diff
@@ -82,7 +82,7 @@ class ATM {
     }
     
     func canWithdraw(value: Int) -> String {
-        return "Can withdraw: \(self.startPile.canWithdraw(value))"
+        return "Can withdraw: \(self.startPile.canWithdraw(v: value))"
```
Comment (ochococo) [inline Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift]: We should avoid parameters like `v`. May I suggest `amount:` ?

File: Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift [hunk 3]
```diff
@@ -249,11 +249,11 @@ var c = IntegerVarExp(name: "C")
 
 expression = AddExp(op1: a, op2: AddExp(op1: b, op2: c)) // a + (b + c)
 
-intContext.assign(a, value: 2)
-intContext.assign(b, value: 1)
-intContext.assign(c, value: 3)
+intContext.assign(integerVarExp: a, value: 2)
```
Comment (ochococo) [inline Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift]: `integerVarExp` -> `expression` ?

File: Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift (hunk 1 above)
Comment (ChristopherBell) [inline Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift]: Of course.

**Comment by ChristopherBell** [comment]
I approve of the changes.

File: Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift [hunk 4]
```diff
@@ -30,30 +30,30 @@ class MoneyPile {
         self.nextPile = nextPile
     }
     
-    func canWithdraw(v: Int) -> Bool {
+    func canWithdraw(amount: Int) -> Bool {
 
-        var v = v
+        var val = amount
```
Comment (ochococo) [inline Design-Patterns.playground/Pages/Behavioral.xcplaygroundpage/Contents.swift]: `val` -> `value` ?

**Comment by ochococo** [comment]
@ChristopherBell 
This file is generated (https://github.com/ochococo/Design-Patterns-In-Swift/blob/master/GENERATE.md) and this change won't help. 
Please change this in source files 🙏

**Comment by Anniywell** [comment]
Great swift3!

**Comment by funky-monkey** [comment]
When will this be merged?

**Comment by ochococo** [comment]
Probably never as I have 80% Swift3 with some changes and refactor locally and will push very soon.

**Pull request closed without merging**